presentation/frontend/src/presentation-frontend/favorite-properties/FavoritePropertiesManager.ts

```diff
@@ -153,25 +156,59 @@ export class FavoritePropertiesManager implements IDisposable {
     propertiesOrder.forEach((oi) => (oi.priority = priority--));
   };
 
+  private isInitialized(imodel: IModelConnection): boolean {
+    const iTwinId = imodel.iTwinId!;
+    const imodelId = imodel.iModelId!;
+    return this._imodelProperties.has(getiModelInfo(iTwinId, imodelId));
+  }
+
+  /**
+   * Checks if {@link FavoritePropertiesManager.initializeConnection} has been called for a given imodel.
+   * Can be removed when {@link FavoritePropertiesManager.has} and {@link FavoritePropertiesManager.sortFields} are removed.
+   */
   private validateInitialization(imodel: IModelConnection) {
     const iTwinId = imodel.iTwinId!;
     const imodelId = imodel.iModelId!;
-    if (!this._imodelProperties.has(getiModelInfo(iTwinId, imodelId))) {
+    if (!this.isInitialized(imodel)) {
       throw Error(
         `Favorite properties are not initialized for iModel: '${imodelId}', in iTwin: '${iTwinId}'. Call initializeConnection() with an IModelConnection to initialize.`,
       );
     }
   }
 
+  /**
+   * Calls {@link FavoritePropertiesManager.initializeConnection} and caches the promise which should be awaited by calling {@link FavoritePropertiesManager.ensureInitialized}.
+   * @internal
+   */
+  public startConnectionInitialization(imodel: IModelConnection) {
+    if (!this.isInitialized(imodel) && !this._imodelInitializationPromises.has(imodel)) {
+      // eslint-disable-next-line deprecation/deprecation
+      this._imodelInitializationPromises.set(imodel, this.initializeConnection(imodel));
+    }
+  }
+
+  /**
+   * Performs the initialization process or finishes the one that was started by {@link FavoritePropertiesManager.startConnectionInitialization}.
+   * @internal
+   */
+  public async ensureInitialized(imodel: IModelConnection) {
+    if (this.isInitialized(imodel)) {
+      return;
+    }
+
+    // eslint-disable-next-line deprecation/deprecation
+    const promise = this._imodelInitializationPromises.get(imodel) ?? this.initializeConnection(imodel);
```

Comment:
should the promise returned by `initializeConnection` be added to `_imodelInitializationPromises` before returning it?

Reply:
Yes, and I figured that we should also remove the promise from the map when it's resolved to free the memory.